### PR TITLE
ACAI-1063: Add callbacks for various events in the scheduler

### PR DESCRIPTION
**Pull request by seangolob**
Add callback support for various events in the scheduler.

- After back and forth with Louis and Josh, this seems like the most appropriate solution to handle click events in the scheduler.
- We are creating 3 callbacks, and cleaning up some other callbacks.
- We leverage `data-analytics-id` to help Jade identify what is get clicked on.
- `onError` is still in progress

### CODE REVIEW

File: clients/scheduler/src/components/time-slot-select/index.tsx [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
-      const availableSlots = newProvider.providerSlots.filter(slot => new Date(slot.start) >= earliestAvailable)
-      const mergedSlots = provider ? [...provider.providerSlots, ...availableSlots] : availableSlots
-      return {providerId: newProvider.providerId, providerSlots: mergedSlots}
-    })
-
-    setProviderTimeSlots(mergedProviderAvailability)
-  }, [providerTimeSlots])
-
-  const {minDate, maxDate} = useMemo(() => {
+  const { fetchTimeSlots, date, setDate, appointmentBufferInMintues } =
```
Comment (seangolob) [inline clients/scheduler/src/components/time-slot-select/index.tsx:16]: The only actual change to this file was removing `daysToFetch`, an unused variable. Other than that, this is just VScode formatter changes.

File: clients/scheduler/src/hooks/app-context.tsx [hunk 2]
```diff
@@ -23,11 +23,10 @@ export const AppContext = createContext<IAppContext>({
   appointmentCoding: {},
   bailoutURL: '',
   callbacks: {
-    onBookingConfirmed: () => {},
+    onClick: () => {},
+    onChange: () => {},
+    onError: () => {},
     onBookingError: () => {},
```
Comment (seangolob) [inline clients/scheduler/src/hooks/app-context.tsx:29]: I will be doing the error callbacks when in a separate PR.

File: clients/scheduler/src/components/time-slot-select/index.tsx [hunk 3]
```diff
@@ -53,48 +78,48 @@ export const TimeSlotSelect = () => {
     const minDate = new Date(slots[0].start)
     const maxDate = new Date(slots[slots.length - 1].start)
 
-    return {minDate, maxDate}
+    return { minDate, maxDate }
   }, [providerTimeSlots])
 
   useEffect(() => {
-    if (typeof maxDate === "undefined" || date >= maxDate) {
+    if (typeof maxDate === 'undefined' || date >= maxDate) {
```
Comment (louisnk) [inline clients/scheduler/src/components/time-slot-select/index.tsx:85]: Do we want to change the quote format across the project? Is this the existing eslint config for the project? 

If we want this to get back into upstream ever, it doesn't seem like we want to go all out on linting to our own code style. If we are cool with just keeping our own fork forever then go all out!

File: clients/scheduler/src/components/time-slot-select/index.tsx [hunk 4]
```diff
@@ -53,48 +78,48 @@ export const TimeSlotSelect = () => {
...
   const dayOfTimeSlots = useMemo(() => {
-    return providerTimeSlots.map((provider) => {
+    return providerTimeSlots.map(provider => {
       return {
         providerId: provider.providerId,
-        providerSlots: provider.providerSlots.filter((slot) => {
+        providerSlots: provider.providerSlots.filter(slot => {
           return isSameDay(new Date(slot.start), date)
-        })
+        }),
```
Comment (louisnk) [inline clients/scheduler/src/components/time-slot-select/index.tsx:96]: same question on code formatting

File: clients/scheduler/src/index.html [hunk 5]
```diff
@@ -21,68 +21,78 @@
...
-
       const callbacks = {
-        onBookingConfirmed: (e) => {
-          console.log("onBookingConfirmed", e)
-        },
-        onBookingError: (e) => {
-          console.log("onBookingError", e)
+        onClick: (e, config) => {
+          if (e && e.target) {
+            // Prescribed method for getting the element clicked
```
[the remark this answers is not in the document]
Comment (louisnk) [reply clients/scheduler/src/index.html:27]: Why not create a wrapper somewhere in the project to help add this to events or something? Or we don't care?

**Review by louisnk — APPROVED** [review]

File: clients/scheduler/src/components/time-slot-select/index.tsx (hunk 3 above)
Comment (seangolob) [reply clients/scheduler/src/components/time-slot-select/index.tsx:85]: I think this code is due to a developer on the canvas side who wasn't setup properly. Nick.... Oh Nick...

FWIW, I don't care about linting at this point in time. When it is more appropriate, we can properly lint this repository. 

Also, the configuration in the package.json suggests single quotes for prettier.

```
"prettier": {
    "semi": false,
    "singleQuote": true,
    "arrowParens": "avoid"
  },
```

**Pull request merged**